src/main/telemetry/smartport.c

```diff
@@ -198,19 +262,29 @@ static void smartPortSendByte(uint8_t c, uint16_t *crcp)
     *crcp = crc;
 }
 
-static void smartPortSendPackage(uint16_t id, uint32_t val)
+static void smartPortSendPackageEx(uint8_t primId, uint8_t* data)
 {
     uint16_t crc = 0;
-    smartPortSendByte(FSSP_DATA_FRAME, &crc);
+    smartPortSendByte(primId, &crc);
+    for(uint8_t i=0;i<6;i++) {
+        smartPortSendByte(*(data++), &crc);
+    }
+    smartPortSendByte(0xFF - (uint8_t)crc, NULL);
+}
+
+static void smartPortSendPackage(uint16_t id, uint32_t val)
+{
+    uint8_t packet[6];
```

Comment:
Generally speaking the packet is the entirety of the thing being sent - that it is a payload enclosed by header and trailer bytes. So I would use the term `payload` here (alternatively `frame` could be used). Better still, use the `struct` that I suggested that you define.